src/eduid_userdb/element.py

```diff
@@ -531,17 +546,28 @@ def _get_primary(self, elements):
         """
         if not elements:
             return None
-        res = [x for x in elements if x.is_primary is True]
+        verified = [x for x in elements if x.is_verified is True]
+        if len(verified) == 0:
+            try:
+                assert len([e for e in elements if e.is_primary]) == 0
+            except AssertionError:
+                raise PrimaryElementViolation('There are unconfirmed primary elements')
+            return None
+        res = [x for x in verified if x.is_primary is True]
         if len(res) != 1:
             raise PrimaryElementViolation("{!s} contains {!s}/{!s} primary elements".format(
                 self.__class__.__name__,
                 len(res), len(elements)))
-        if self.require_verified_primary and not res[0].is_verified:
-            raise PrimaryElementViolation("Primary element of {!s} must be verified".format(
-                self.__class__.__name__,
-            ))
         return res[0]
 
+    @property
+    def verified(self):
+        """
```

Comment:
is it worth disambiguating 'verified' even more from 'is_verified' by calling this function 'verified_elements', or is that too verbose?